VirtoCommerce.CatalogModule.Web/Scripts/blades/bulk/action-list.js

```diff
@@ -26,5 +31,31 @@ angular.module('virtoCommerce.catalogModule')
         $scope.blade.title = "Bulk action list";
         $scope.blade.subtitle = "Select action for bulk operation";
 
+        blade.initializeAction = function (action) {
+            action.title = `actions.types.${action.name}.title`;
+            action.subtitle = `actions.types.${action.name}.subtitle`;
+            action.id = action.name;
+            action.icon = 'fa fa-cogs';
+            action.actionDataContext = {
+                actionName: action.name,
+                contextTypeName: action.contextTypeName,
+                dataQuery: {
+                    dataQueryTypeName: action.dataQueryTypeName,
+                    categoryIds: _.pluck(blade.selectedCategories, 'id'),
+                    objectIds: _.pluck(blade.selectedProducts, 'id'),
+                    catalogIds: [blade.catalog.id]
+                }
+            };
+
+            if (action.name === 'ChangeCategoryBulkUpdateAction') {
```

Comment:
This should be done in registrar, we should not write this action specific logic.